Climbing IA

kfloey opened the pull request:
**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Climbers

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
https://duck.co/ideas/idea/4759/hiking-and-climbing-spots-near-a-location

**Which existing Instant Answers will this one supercede/overlap with?**
None

**Are you having any problems? Do you need our help with anything?**
There was very little information on longtails, so I need help formatting the output.  So far, all I have is the crawler that parses data from the entire site.  The data is then transfered into a JSON document.

https://duck.co/ia/view/climbing

brendanheywood commented:
Can you please add a user agent, either the same as what the normal duckduck go crawl would use or something similar?

jdorweiler commented:
@kfloey Great, thanks for working on this :+1: 
Just a few comments for now.  I'll give it a better look tomorrow. 

Example of the DDG user agent: https://github.com/duckduckgo/zeroclickinfo-longtail/blob/df144793ed893b204b10a392def1e995b5aa5aec/share/longtail/quora/download.js#L50

Is there a way to limit it to a single area for now?

kfloey commented:
I will get the user agent added in the next commit.  The crawler can be limited to one area by changing the URL from world to that area. The crawler will stop at the first crag, so just start it at any crag's URL.  I can change the URL to one spot if that's what you are looking for.

jdorweiler commented:
@kfloey thanks, I ran it on my local area and got this https://gist.github.com/jdorweiler/8542e72be8b84a2fcf4c
 
I think we will have to have separate crag and area entries for this to work.  The crag would look something  like this:
```
"Shawangunks": {
        "number of routes": "1,706", 
        "styles": {
            "Aid": ".3%", 
            "Boulder": "17%", 
            "Sport": ".1%", 
            "Top Rope": ".6%", 
            "Trad": "11%", 
            "Unknown": "70%"
        }, 
        "type": "crag", 
        "url": "http://www.thecrag.com/climbing/united-states/shawangunks"
        "location":  lat/lon location here?
}
```
And the area might look like this:
```
 "The Trapps": {
                "height": "140ft", 
                "link": "/climbing/united-states/shawangunks/area/12665509",  
                "routes": "953", 
                "ticks": "5,397",
               "location":  lat/lon location here ?
            }, 

```

The actual output for the longtail to work will have to be something like this:

```
 <doc>
     <field name="title"><![CDATA[Shawangunks]]></field>
     <field name="paragraph"><![CDATA[
            "number of routes": "1,706"             
            "Aid": ".3%", 
            "Boulder": "17%", 
            "Sport": ".1%", 
            "Top Rope": ".6%", 
            "Trad": "11%", 
            "Unknown": "70%"
           ]]>
      </field>
     <field name="source"><![CDATA[climb]]></field>
     <field name="meta"><![CDATA[
        {
           "url":" link back to the page you want to show,
          "location":  { lat: #### ,  lon: ### }
          "}
        ]]></field>
</doc>

then repeat this format for the areas too
```
I just copy pasted your json into the paragraph field but it will have to be plaintext in the same format as you want it displayed on the site.  The meta field does need to be valid json though.  

There are a bunch of other fields we can play around with later but this should be enough for me to get it working on a test server for us.

jkanarek commented:
It'd be great to expose the location of the routes in the IA. A lot of their summaries include that information as does the lat/long field.

brendanheywood commented:
A few thoughts:

* Shawagunks isn't a great example because it has little or no description text filled out. You probably want to use the 'summary' field in the IA paragraph rather than just a bunch of stats. If no summary then possible truncate part of the description as well. 

Compare it to a crag that has this info eg these examples:

http://www.thecrag.com/climbing/australia/gara-gorge/upper-gara-gorge
http://www.thecrag.com/climbing/australia/arapiles

* Almost all crags have location boundaries which are a bit more useful than the centroid lat/ling which is what gets pushed into the page metadata. Do you want the boundary instead?

```
<meta property="place:location:longitude" content="-74.31680830844" />
<meta property="place:location:latitude" content="41.62497737531" />
```

The boundary is currently there but tangled in JavaScript, if there is a preferred way for me to mark this up let me know, and I'll make the javascript then read from this instead. I'm thinking about making this a data attribute on the ```<li>``` tag in the navigation structure.

```
var boundary = {
id: '11904013', type:'Crag',name:'Shawangunks',geometry:[[-74.2347193,41.9804434],[-73.8172389,41.8660035],[-74.4599391,41.2370498],[-74.7840357,41.4061931],[-74.2347193,41.9804434]],cen: ["-74.31680830844","41.62497737531"]
};
```

* Routes on the other hand while they can be located are very rarely located because it a very small diminishing return. They inherit their location from their parent node, and given that most cliffs or areas are ~ 100m wide there is little point.

kfloey commented:
@jdorweiler do we want to include the areas within each main crag as a separate output text?  I scraped the data about the subareas from the main crags, but did not crawl each subareas page.  Right now, if the crawler scraped from the "world" page, it would collect data from each page, including the areas under it.

@brendanheywood I noticed the center location in the metadata, and will look at parsing for the boundaries.  Would scraping the boundary be used for reproducing the map? Also, what would you suggest to display when there is neither a summary or description?  Lastly, when you mentioned the parsing until the first crag in the other forum, is this strictly to areas labeled crag or any area with subareas and not routes?

brendanheywood commented:
It all depends on what you want to display. The boundary is pretty useless unless you stick it on a map. Would you want to render a map from say openstreetmap with a marker in the AI? I'm guessing you aren't going to want to use a google map :)

If you aren't going to render it then don't scrape it. At least in text a simple lat long can be easily linked and copied etc

Re: when there is no summary or description: internally we have a bit of similar logic for making the meta description which pieces together a sentence when bits are missing. It may be easier to piggy back on top of this and just scrape the meta description so you don't have to worry about any of this logic. It does have a fixed boiler plate prefix you' want to remove:

```
<meta property="og:description" content="Rock climbing Info • Forums • Photos • Ticking •  Cape Raoul, Australia: Amazing wild location" />
```

Ie you'd just want the "Cape Raoul, Australia: Amazing wild location"

It might be more useful to instead of working from the data out, work from the IA back in: what is the anatomy of a perfect answer? Then it's much easier to figure out what fields to pull from, or what data is missing that we need to supply

Re: 'parsing until the first crag' - this is sort of up to you. Internally we have a Top Level Crag concept and in most cases this is I think the most back for buck for the IA's. a TLC is defined as any node whose parent is a region, and it isn't a region. So it could be a 'crag' or could even be a cliff or boulder or whatever. We have a couple of heuristics to help people decide between a 'crag' or 'region' or 'cliff', the simplest is Where do you park the car? If two places use the same car park they are probably cliffs under one crag. If two places have different car parks they are probably different crags. Regions are generally perfect subdivisions of larger regions (eg national, state, council) and adjacent to each other where as crags are often distant from each other.

You are welcome to scrape lower but then you will end up with a much larger number of decreasingly useful things like 'left side' ' rightside' etc. I'd probably start with just the TLC's and then slowly drill down more keeping the signal to noise level high

jdorweiler commented:
@kfloey I think we should get the main crag and then the larger sub-areas.  I agree that going lower than that might just give us lots of not very useful data.  
For the location I think just the lat/lon location is good for now.  Once we have some test data I can see if adding a bounding box will be possible.

kfloey commented:
@jdorweiler I just added another commit.  I didn't create a separate output for crags and their subparts, but I think the output we have now is a good starting place and we will add more after.

Is it possible to add more links to the longtail other than to the region displayed on the IA?

jdorweiler commented:
@kfloey that looks better.  I just had to make a few changes to the output format but I was able to get the test data working :+1: 

I'll ook it over and get back to you with some comments and put it on a test server.

![selection_338](https://cloud.githubusercontent.com/assets/1882892/5951168/40188b58-a731-11e4-8175-5253db45187a.png)

jdorweiler commented on share/longtail/climbing/crawler.pl:
```diff
@@ -0,0 +1,147 @@
+
+my $site = "http://www.thecrag.com";
+my $start = "http://www.thecrag.com/climbing/world";
+my $test = "http://www.thecrag.com/climbing/australia/grampians";
```
Thanks this one has some good data.

brendanheywood commented:
Looks great guys :)

Can I ask the you use 'theCrag' or 'thecrag.com' instead of 'TheCrag' just to keep things consistent.

kfloey commented:
@jdorweiler this may be far ahead of what we are focusing on now, but would we be able to have an output like the one below for queries like: "climbing near me" or "bouldering areas in New York"?

![example](https://cloud.githubusercontent.com/assets/9901144/5965373/a8eb294a-a7c9-11e4-88cc-ed29bc1a502f.png)

jdorweiler commented:
@kfloey yeah I would like to see that too.  Right now we can't do geo search from a longtail but I'm going to work on adding it.

jdorweiler commented on share/longtail/climbing/crawler.pl:
```diff
@@ -0,0 +1,147 @@
+        ]]></field>
+</doc>
+EOH
+;
```
Can you format the output file so it's all in a single line other than the paragraph field?  It's easier to read that way.

jdorweiler commented:
@kfloey I've been playing around with the output data and trying to get geo search working.  I've had some success and I'll hopefully have something on a test server for you soon.  For now can you update your output file to fit with the changes I made?
![selection_343](https://cloud.githubusercontent.com/assets/1882892/6008861/cfcc2bdc-aaf2-11e4-9d7d-1385a006881e.png)

`geo` - Put the lat, lon coordinates in here
`l2_sec_match2` - I'm not quite using this yet but I want to try and add relevant keywords here.  climb/climbing seems okay for now.

jdorweiler commented on share/longtail/climbing/crawler.pl:
```diff
@@ -0,0 +1,147 @@
+		
+		print OUT <<EOH
+<doc>
+     <field name="title"><![CDATA[$place]]></field>
```
Could you append the word "Climbing" to all the titles?  That's going to help us a lot with the relevancy.

jdorweiler commented:
It's pretty rough but I put the test data on here:
https://ddh10.duckduckgo.com/?q=grampians+climbing&ia=thecrag

It currently only has the Grampians and Safe Harbor (craig near me for geo testing).  Geo search is somewhat working but I gave it a huge 20k km radius just for now.  I'll post again when I've made some more progress with it.

I still need to update the tab name but I did see your comment about that @brendanheywood

brendanheywood commented:
Re adding the word 'climbing' to all entries, I'd not hard code this and instead look it up from the page title. Some areas will be 'blah trad climbing' and some will be 'blah ice climbing' and others will be 'blah bouldering', and if the area is unknown it defaults to 'blah climbing'. In particular 'bouldering' will be a much better match where applicable.

It's in the markup in both titles:
```
<title>North Grampians, Rock Climbing | theCrag</title>
```
or 
```
<meta property="og:title" content="North Grampians, Rock Climbing" />
```
vs
```
<meta property="og:title" content="Andersens, Bouldering" />
```

kfloey commented:
@jdorweiler Fixed the output file, and its looking good so far.  Could a similar map to the one on the site be added to the IA?  If so, I could scrape the boundary from the JS.  Also, if there is anyway I can help with adding that feature let me know.

@brendanheywood will do.

jdorweiler commented:
@kfloey Thanks!  I'm not 100% sure how or if we can use a map.  It would use a javascript function on our frontend called `DDG.duckbar.add_map()` though.  Here's an example https://duckduckgo.com/local.js?q=new%20york,  you can copy/paste that call into the console to open up a map.  We can probably use the crag bounding box data to do something similar.  I'm not sure exactly what data we need or what the format is for the add_map function so let me know if you figure it out.

kfloey commented:
@jdorweiler from first glance it seems as if the points that are used in the crag bounding box can be plugged into the format for DDG. I'll look at it more and let you know.
Also, for clarity, which console can I paste the JS into to test?

jdorweiler commented:
@kfloey it should work with your browser console.

jdorweiler commented:
@kfloey Let's just work on getting the correct data for now.  I think that way we can move this along and have it working just as a normal longtail (text only search).  Then as long as the location data is there I can try to get the geo/maps working without everyone waiting around.   

What other data should we show other than the description?  I think the number of routes would be good to see.

kfloey commented:
@jdorweiler Ok, I got some of the js worked out for adding the map, but I agree we should finalize the normal longtail. The code was scraped from the page and would give you a map of the Grampians, but doesn't have a label yet.

```javascript
DDG.duckbar.add_map([{"licence":"Data \u00a9 OpenStreetMap contributors, ODbL 1.0. http:\/\/www.openstreetmap.org\/copyright","osm_type":"relation", "polygonpoints":[[142.3673301,-36.8786611],[142.3635536,-36.8878606],[142.3709673,-36.9664513],[142.3877221,-36.9812302],[142.3907003,-37.0791231],[142.3228158,-37.0859679],[142.2730358,-37.1084902],[142.2594764,-37.1162237],[142.2594764,-37.1162237],[142.1432537,-37.112225],[142.1525234,-37.0464882],[142.1023554,-37.0006956],[142.0730764,-37.0274242],[142.0078987,-37.1487151],[142.0422739,-37.1882048],[142.0938152,-37.1882988],[142.1409867,-37.115715],[142.2601424,-37.120566],[142.2646648,-37.1665256],[142.2353107,-37.2411782],[142.1993638,-37.3654077],[142.1862156,-37.5232583],[142.236169,-37.5412273],[142.280665,-37.4576598],[142.3621328,-37.5043359],[142.3048837,-37.6263896],[142.3159559,-37.6450134],[142.3729474,-37.6118407],[142.4841153,-37.4804027],[142.5675811,-37.3878186],[142.6359327,-37.3085251],[142.6057203,-37.2112482],[142.5743502,-37.1690797],[142.731682,-37.1571102],[142.7487353,-37.1703154],[142.7643565,-37.172504],[142.7698497,-37.1433635],[142.7505647,-37.1341225],[142.7290211,-37.1495984],[142.5701483,-37.1634296],[142.5298435,-37.1240031],[142.4980503,-37.0235263],[142.4714857,-36.9833251],[142.4195581,-36.9021038],[142.3998605,-36.8779129],[142.3673301,-36.8786611]], "display_name":"Grampians, North West, Victoria, Australia", "lat":-37.26530166539, "lon":142.37633846546, "class":"boundary", "icon":"http:\/\/open.mapquestapi.com\/nominatim\/v1\/images\/mapicons\/poi_boundary_administrative.p.20.png"}])
```

As for the rest of the longtail, i think the number of routes and types of climbing as @brendanheywood mentioned would be a good addition.  Also, should i shorten the amount of text extracted from the summary.  Looking at the test server, the Grampians overview seems a little crowded with information.

jdorweiler commented:
@kfloey Nice map!  We have some stackoverflow posts that take up most of the page so the length here isn't a big deal.  I think visually it would be better to keep it to a paragraph though.  Is there an easy way for you to do that?

kfloey commented:
Yeah, I can shorten how many paragraphs I extract from the description.  The summaries are short, so I originally scraped it and 2 paragraphs of the description, but I can change it to one.

Also, in the commit I just made, the js for each area is output into an maps.js file.

jdorweiler commented:
@kfloey You will have to append that to the paragraph field and separate new lines with a `<br>` tag.

kfloey commented:
@jdorweiler I added the extra information and `<br>` tags.  Is there any other markup that could be done to improve the IA?

jdorweiler commented:
@kfloey Thanks!  You can only use `<br>` tags.  Any other html will be stripped off.

kfloey commented:
@jdorweiler While you are working on geo location, is there something that could be added similar to `l2_sec_match2` that would connect the answers of searches like "climbing in North Carolina" to the crags that fall under that region.  We could connect each output with the area directly above it using the "breadcrumbs."

 Could we also use the "Classic Crags" ratings somehow? Look at http://www.thecrag.com/climbing/united-states/north-carolina for a decent example.

For priority on which areas to show first, should there be a label in the metadata or somewhere that signifies that an area is the "top level crag" or sub-area?


Lastly what should be added for searches like "bouldering in North Carolina" or "Trad climbing near [place]" that specify the type of climbing?

jdorweiler commented on share/longtail/climbing/crawler.pl:
```diff
@@ -0,0 +1,158 @@
+<field name="paragraph"><![CDATA[$paragraph]]></field>
+<field name="source"><![CDATA[climb]]></field>
+<field name="meta"><![CDATA[{"url":"$url"}]]></field>
+<field name="geo">$lat, $lon</field>
```
Can you move this back into the meta field since we're not using it yet?

jdorweiler commented on share/longtail/climbing/crawler.pl:
```diff
@@ -0,0 +1,158 @@
+		if($#descriptioninfo == 0){
+			my @description = $descriptioninfo[0]->findvalues('div/div/p');
+			$paragraph .= ($description[0] . "<br>");
+			if($#description>0) {$paragraph .= "...";}
```
Can you make sure that the ellipsis falls on the same line?  
![selection_345](https://cloud.githubusercontent.com/assets/1882892/6107779/b5427e26-b03b-11e4-922e-150ef2fb0740.png)

jdorweiler commented:
@kfloey I answered your comments below. 

```connect the answers of searches like "climbing in North Carolina" to the crags that fall under that region. We could connect each output with the area directly above it using the "breadcrumbs."```

We could try this with a link in the description that links back to the parent page on thecrag.  So each sub-area could have a link back to the parent.  Could we also link a parent crag back to a page with other crags in that area too?  This could be another line under the number of routes "Region: [link to region]"

```Could we also use the "Classic Crags" ratings somehow? Look at http://www.thecrag.com/climbing/united-states/north-carolina for a decent example.
For priority on which areas to show first, should there be a label in the metadata or somewhere that signifies that an area is the "top level crag" or sub-area?```

We could easily do some sorting based on any measure of popularity.  The easiest way for this to work would be for us to pre-compute a popularity score and add it as another field.  Computing the score might be tricky though since it will have to be the result of some function that considers several attributes of the area (routes, classic, ... ).   

```lastly what should be added for searches like "bouldering in North Carolina" or "Trad climbing near [place]" that specify the type of climbing?```

Lets push these cases off for now and revisit them as a separate issue later.

kfloey commented:
@jdorweiler I fixed some of the formatting for the IA, and added both the number of ascents and type of area to the meta field.  

When considering these ideas:
>We could easily do some sorting based on any measure of popularity.

If the answers are already sorted by the user's search, it would probably be simple enough just to rate the places on the number of ascents they have.  Checking a few examples, the most documented, and better areas have more ascents, but we can always base the popularity score off of more than that. We should especially make sure that the main crags come before the subareas, and that information is in the meta field as well.

>We could try this with a link in the description that links back to the parent page on thecrag. So each sub-area could have a link back to the parent. Could we also link a parent crag back to a page with other crags in that area too? This could be another line under the number of routes "Region: [link to region]"

I added the region to the description in the format `Region: $parent [$parenturl]` is there a way to make the name of the region above the crag a link itself instead of having the long url beside it?

jdorweiler commented:
@kfloey Thanks, this is looking really good.

>If the answers are already sorted by the user's search, it would probably be simple enough just to rate the places on the number of ascents they have. Checking a few examples, the most documented, and better areas have more ascents, but we can always base the popularity score off of more than that. We should especially make sure that the main crags come before the subareas, and that information is in the meta field as well.

I think that using the most ascents sounds like a good approach.  As long as we have some popularity score in the data then I can try to adjust things on our back-end to deal with subareas.  

>I added the region to the description in the format Region: $parent [$parenturl] is there a way to make the name of the region above the crag a link itself instead of having the long url beside it?

Sorry I was thinking of just using a link tag like this:
![selection_346](https://cloud.githubusercontent.com/assets/1882892/6128411/8cce2d00-b103-11e4-9c60-0b6a5aecf8df.png)

jdorweiler commented on share/longtail/climbing/crawler.pl:
```diff
@@ -0,0 +1,170 @@
+		print OUT <<EOH
+<doc>
+<field name="title"><![CDATA[$place $styleinfo]]></field>
+<field name="12_sec_match2">climb climbing</field>
```
`l2_sec_match` not `12_sec_match`

jdorweiler commented on share/longtail/climbing/crawler.pl:
```diff
@@ -0,0 +1,170 @@
+<field name="paragraph"><![CDATA[$paragraph]]></field>
+<field name="source"><![CDATA[climb]]></field>
+<field name="meta"><![CDATA[{"url":"$url", "lat":"$lat", "lon":"$lon", "ascents" ="$ascents", "type"="$typelabel" }]]></field>
+</doc>
```
Could you add a `</add>` tag as the last line of the doc too?

kfloey commented:
@jdorweiler I'm still not totally sure which parts of the IA can only be done on the back-end, so is there somehow I can add the link tag, or is that on your end?

brendanheywood commented:
On the popularity front we already have an internal metric for each route which takes into account the rating that each climber gives a route, and not just total ascents. This is much better as it removes the bias for easy routes that get a lot of traffic vs better quality but harder routes, or routes which are more accessible etc. 

This is for individual routes, eg:
http://www.thecrag.com/climbing/australia/nowra/the-grotto-cliffline/route/13231195

![image](https://cloud.githubusercontent.com/assets/187449/6147405/46f0e6bc-b24b-11e4-817e-e52a5cce7bb2.png)

Then for whole crags we have an aggregate metric which takes this into account, along with the number of ascents and other metrics like many different people have climbed there to produce a 'crag kudos' score. This isn't really exposed much but we do use it eg to show the top 5 crags on certain pages. I think it's better if we just add this as a data attribute somewhere you can parse and use directly.

I've logged an issue on our side for this, it should be done in our dev system shortly and I'll get you guys to swap over to scraping that until our next release
https://github.com/theCrag/website/issues/1853

btw our dev site is: http://dev.thecrag.com/ Please don't hammer it too much as it's on a small domestic internet connection

jdorweiler commented:
@kfloey you can add the link to the paragraph text as a normal link `<a href="....">North West</a>`. 
@brendanheywood Thanks! The aggregate popularity would be good to have. 

Just to update I think the plan from here looks something like this:
 - Add the link to the paragraph field
 - Add the aggregate popularity field that @brendanheywood mentioned once it's ready
 - I'll scrape a larger area and put the data on a test server
 - adjust back end, test relevancy, and repeat until we are happy
 - design feedback

kfloey commented:
@jdorweiler Thanks!  I was just a little unsure of what tags could be used, but the link is added now.  

In reference to the design of the IA, do we want the title to include the type of climbing?  That is what we initially decided, but it seems a little redundant to have the title be `Grampians Trad Climbing` if the first set of information listed is `Mostly trad climbing`.

Also, is there any visual that we would want to / could include in the IA.  If we are unable to add a map, would it be beneficial to add one of the photos for the area?

brendanheywood commented:
> would it be beneficial to add one of the photos for the area?

We are working on allowing a 'cover photo' for an area. In the mean time you can use the og:image if you want, and when we've implemented the cover photo the og:image will point at this. Currently the og:image is either a topo, or failing that a random photo from the area, or failing that our logo. YMMV.

jdorweiler commented:
@kfloey lets leave the type of climbing off from the title.  I think you're right and it does look a little redundant.
We don't have a way to use image unfortunately.  At some point I hope that we can use custom templates for a longtail similar to how spice uses them.

kfloey commented:
@jdorweiler 
>At some point I hope that we can use custom templates for a longtail similar to how spice uses them.

Same here.  Because we can only add text, I feel as if the formatting of the output for the IA is complete for now.  Do you or @brendanheywood think there is anything else we should add?

brendanheywood commented:
Where can I see the latest IA in action? Apart from the crag quality score, is there any other data you like me to add to make the scraping easier? I'd prefer it if you guys were using data from data attributes or meta tags or similar as those are pretty stable and it's unlikely that we'll accidentally break them in the future. Things like parsing the GPS data from json inside a script tag will be much less stable, so anything like that let me know.

jdorweiler commented:
After we get the quality score I'll scrape a larger area and put the data up for testing.  Other than that I think we have everything I can think of for now.

kfloey commented:
@brendanheywood I would assume most of the data comes from stable attributes.  The crawler is currently programmed to collect the GPS data from the json, but we haven't found any use for it, so I wouldn't consider worrying about that yet.

jdorweiler commented:
Any update on adding the popularity field?

brendanheywood commented:
Sorry no, this isn't my code but I've nudged Simon who looks after this. I should have something at least on our dev box in a couple days.

brendanheywood commented:
hi @jdorweiler this is now live in our prod system eg:

http://www.thecrag.com/climbing/australia/cape-raoul

```<body   ....  data-nid="11744779" data-kudos="557">```

kfloey commented:
@brendanheywood thanks!
@jdorweiler the kudos is now scraped and added to the meta field

jdorweiler commented:
Thanks!  @brendanheywood should I still use your dev site for scraping?

brendanheywood commented:
No that code is live now so using prod is good to go

Sent from my iPhone

On 27/02/2015, at 1:57 AM, Jason <notifications@github.com> wrote:

Thanks! @brendanheywood <https://github.com/brendanheywood> should I still
use your dev site for scraping?

—
Reply to this email directly or view it on GitHub
<https://github.com/duckduckgo/zeroclickinfo-longtail/pull/12#issuecomment-76191718>
.

jdorweiler commented:
Just and update.  I did a small scrape of ~200 pages and still need to update some of the backed to get this to show correctly.  I'll let you know when it's ready but it should be sometime in the next few days.

jdorweiler commented:
![selection_376](https://cloud.githubusercontent.com/assets/1882892/6463997/26f31738-c185-11e4-8cd5-b56cf989587c.png)

Here's a list of all the titles I scraped.  https://gist.github.com/jdorweiler/6fe61be0e38f3796b400

It's live on ddh10: https://ddh10.duckduckgo.com/?q=central+grampians&ia=climb

jdorweiler commented:
![selection_377](https://cloud.githubusercontent.com/assets/1882892/6464206/b42ea24c-c186-11e4-94e4-d170c0c51d17.png)

brendanheywood commented:
Yeah ideally would be thecrag.com, thecrag by itself looks like a typo.

jdorweiler commented:
@brendanheywood good point.  I updated the moreAt link name.

jdorweiler commented:
Sorry for the delay.  I'm back with some design feedback.  

Just one issue with some descriptions ending with an ellipsis. 
![selection_405](https://cloud.githubusercontent.com/assets/1882892/6696769/1ff9d028-ccc2-11e4-9d80-74ebd60361d1.png)

Could those just end with a period?  Everything looks great otherwise.

brendanheywood commented:
Wouldn't this be valid / expected for when you are truncating the description you are parsing from our end?

This particular case is a little tricky as you are truncating the description, and the cut just happens to also end at the end of a sentence, or is this something you are aiming for on your end?

> Heading towards Arapiles for some legendary climbing and starting to see it clearly for the first time, you will not have been alone if you feel a little disappointed. For taken from a distance Arapiles looks like a pile of choss.
>
> It is the closer inspection that yields a glimpse of the true magic of Arapiles. For this is when Arapiles reveals itself. The beautiful rich golden and gray sandstone, hard and solid. The labyrinth of small gullies and hidden faces, the technical and varied climbing.

Wouldn't the ellipsis in these cases make more sense showing that there is more description omitted?

kfloey commented:
@brendanheywood Because of how the data is parsed, the description is truncated after the first paragraph, so it would probably happen at the end of a sentence each time.

@jdorweiler I added the ellipses to indicate that there was more information.  Is there any other ways to indicate that that might be less confusing after a paragraph?

brendanheywood commented:
How would this work if there was only 1 really long paragraph, in that case
is there also a second hard character limit and truncated mid sentence? We
can't guarantee much about the shape of this data

Sent from my iPhone

On 19/03/2015, at 1:57 AM, Kevin Foley <notifications@github.com> wrote:

@brendanheywood <https://github.com/brendanheywood> Because of how the data
is parsed, the description is truncated after the first paragraph, so it
would probably happen at the end of a sentence each time.

@jdorweiler <https://github.com/jdorweiler> I added the ellipses to
indicate that there was more information. Is there any other ways to
indicate that that might be less confusing after a paragraph?

—
Reply to this email directly or view it on GitHub
<https://github.com/duckduckgo/zeroclickinfo-longtail/pull/12#issuecomment-83009339>
.

jdorweiler commented:
What if we just have a cut-off for the number of sentences that are allowed?  Maybe 8 sentances max.  If the description was cut-off then add an ellipsis.

abeyang commented:
Hi, just a few quick design feedbacks. First, when it's in tile view, the attribution icon is cut-off. Is it possible to add this css, specific for this IA, as follows:

```css
.tile__more-link {
  padding-top: 0.1em
}
```

Second, any way we can reduce the top/bottom padding inside the detail pane? It's humungo when there's not that much content:

![pasted_image_3_20_15__1_53_pm](https://cloud.githubusercontent.com/assets/794084/6760695/8e48e292-cf08-11e4-9370-af64db2259ec.png)

jdorweiler commented:
Sorry but without custom templates we can't make any css changes. 
@brendanheywood did you still want to provide a data dump or should we scrape the site?

brendanheywood commented:
I think the scrape is fine. And it's already working :)

Sent from my iPhone

On 22/03/2015, at 6:09 AM, Jason <notifications@github.com> wrote:

Sorry but without custom templates we can't make any css changes.
@brendanheywood <https://github.com/brendanheywood> did you still want to
provide a data dump or should we scrape the site?

—
Reply to this email directly or view it on GitHub
<https://github.com/duckduckgo/zeroclickinfo-longtail/pull/12#issuecomment-84448044>
.

jdorweiler commented:
@brendanheywood @kfloey This is ready to go I'm just waiting on an internal PR to go with this.

kfloey commented:
@jdorweiler Great! Is it still running on ddh10 because it doesn't seem to be working right now?

jdorweiler commented:
@kfloey no I took it off ddh10.  You don't need anything special in the readme.  It's more for install directions in cases where someone uses an unusual library.

jdorweiler commented:
@kfloey @brendanheywood thanks for the hard work on this!  Everything is ready to go so I'm going to merge and start scraping.  I'll let you know when it's live.

brendanheywood commented:
Woot! :)

jdorweiler commented:
Finally live!  https://duckduckgo.com/?q=shawangunks+climbing&ia=climb
It's a little tough to get it to show though.  We had this same issue with the fuel economy longtail too but I've got a plan to fix it.

@kfloey I couldn't get the scraper to run from the "world" level.  I manually ran it for each area (North America, Europe, Australia  .. ) and concatenated the results.

foggg7777 commented:
6665652
